### PR TITLE
String constants session state

### PR DESCRIPTION
**Pull request by mschwoer**
Introduce string constants for easily navigating session state .. replaces https://github.com/MannLabs/alphapeptstats/compare/main...bookkeeping?expand=1

### CODE REVIEW

File: alphastats/gui/utils/options.py [hunk 1]
```diff
@@ -5,130 +5,133 @@
...
         },
         "Volcano Plot": {
             "between_two_groups": True,
-            "function": state.dataset.plot_volcano,
+            "function": dataset.plot_volcano,
         },
-        "Clustermap": {"function": state.dataset.plot_clustermap},
-        # "Dendrogram": {"function": state.dataset.plot_dendrogram},  # TODO why commented?
+        "Clustermap": {"function": dataset.plot_clustermap},
+        # "Dendrogram": {"function": state[StateKeys.DATASET].plot_dendrogram},  # TODO why commented?
```
Comment (JuliaS92) [inline alphastats/gui/utils/options.py:76]: The dendrogram is rather redundant as it is part of clustermap output, but that's just my guess.

File: alphastats/gui/utils/options.py [hunk 2]
```diff
@@ -5,130 +5,133 @@
 from alphastats.loader.SpectronautLoader import SpectronautLoader
 from alphastats.loader.GenericLoader import GenericLoader
 from alphastats.loader.mzTabLoader import mzTabLoader
+from alphastats.gui.utils.ui_helper import StateKeys
 
 
 def plotting_options(state):
+    dataset = state[StateKeys.DATASET]
```
Comment (JuliaS92) [inline alphastats/gui/utils/options.py:12]: Should we maybe just pass dataset and metadata_columns directly instead of the whole state?

**Review by JuliaS92 — APPROVED** [review]
I added some that were missed.
Check the added todo please.

**Comment by mschwoer** [comment]
thanks for adding the ones I missed .. I would leave the TODO you introduced for now, I added another one on `interpretation_options` in another branch..

File: alphastats/loader/BaseLoader.py [hunk 3]
```diff
@@ -61,7 +61,7 @@ def _check_if_columns_are_present(self):
                 "MaxQuant Format: http://www.coxdocs.org/doku.php?id=maxquant:table:proteingrouptable"
             )
 
-    def _read_all_columns_as_string(self):
+    def _read_all_column_names_as_string(self):
```
Comment (JuliaS92) [inline alphastats/loader/BaseLoader.py:64]: Sorry, I meant to do this on a different branch.

File: alphastats/gui/utils/options.py (hunk 2 above)
Comment (mschwoer) [reply alphastats/gui/utils/options.py:12]: for now this is fine I guess

**Pull request merged**